bazel/foreign_cc/BUILD

```diff
@@ -159,30 +160,23 @@ envoy_cmake_external(
     cache_entries = {
         "YAML_CPP_BUILD_TESTS": "off",
         "YAML_CPP_BUILD_TOOLS": "off",
-        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
-        "CMAKE_CXX_COMPILER_FORCED": "on",
```

Comment:
Don't delete this line, this break ASAN.

Reply:
Agreed w.r.t. failing to override the compiler; pushed update.

Need to evaluate the utility of overriding/appending yaml's cmake build
schema flags on top of bazel's preferred flags. Looking to the results of
the next CI evaluation for any remaining steps (and evaluating how the
envoy_cmake_external needs to be further refined.)

Reply:
It would be my preference to apply the CMAKE_CXX_COMPILER_FORCED globally on all dependencies of envoy binary, so that there is one fewer moving parts in the build/link process. But seems that aught to be a subject for a second PR discussion? It appears this clears up all linux-specific build breakage. Seems safe now to proceed with merging (the variant relwithdebinfo toggle seems to be unneeded, we can look at a toggle to vary that at the bazel rules_foreign_cc cmake_external() level in the future.)

Reply:
+1 doing that globally, can you raise another PR for that?

Reply:
It was simpler to establish a new tracking issue on open issues, see https://github.com/envoyproxy/envoy/issues/8554. Once we are confident in such a PR we will escalate the proposal.